src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
+Animal.alive = [];
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    Animal.alive.push(this);
+  }
+
+  bite(obj) {
+    if (obj instanceof Herbivore && obj.health >= 50 && obj.hidden === false) {
```

Comment:
```suggestion
    if (obj instanceof Herbivore && obj.hidden === false) {
```

if creature's health < 50, it will be deleted from array, that's why you don't need this check inside 'if' statement